models/src/main/protobuf/CasperMessage.proto

```diff
@@ -12,19 +12,23 @@ option (scalapb.options) = {
 
 // --------- DeployService  --------
 service DeployService {
-  rpc DoDeploy(DeployString) returns (DeployServiceResponse) {}
+  rpc DoDeploy(DeployData) returns (DeployServiceResponse) {}
   rpc createBlock(google.protobuf.Empty) returns (MaybeBlockMessage) {}
   rpc addBlock(BlockMessage) returns (DeployServiceResponse) {}
   rpc showBlock(BlockQuery) returns (BlockQueryResponse) {}
   rpc showBlocks(google.protobuf.Empty) returns (BlocksResponse) {}
 }
 
-message DeployString {
+message DeployData {
   bytes  user         = 1; //public key
   string term         = 2; //rholang source code to deploy (will be parsed into `Par`)
   int64  timestamp    = 3; //millisecond timestamp
   bytes  sig          = 4; //signature of (hash(term) + timestamp) using private key
   string sigAlgorithm = 5; // name of the algorithm used to sign
+  string from         = 6; //purse address which will be used to pay for the deployment
+  int32 phloPrice     = 7; //phlo price
+  int32 phloLimit     = 8; //phlo limit for the deployment
+  int32 nonce         = 9; //nonce of the deployment
```

Comment:
Comment should be more clear: "nonce for transaction made against `from` wallet"